packages/seacas/libraries/ioss/src/Ioss_ChangeSet.h

```diff
@@ -0,0 +1,101 @@
+// Copyright(C) 2024 National Technology & Engineering Solutions
+// of Sandia, LLC (NTESS).  Under the terms of Contract DE-NA0003525 with
+// NTESS, the U.S. Government retains certain rights in this software.
+//
+// See packages/seacas/LICENSE for details
+
+#pragma once
+
+#include "Ioss_DatabaseIO.h"
+#include "Ioss_DBUsage.h"
+#include "Ioss_Region.h"
+#include "Ioss_ParallelUtils.h"
+
+#include <utility>
+#include <climits>
+#include <cstddef>
+#include <iostream>
+#include <sstream>
+#include <iomanip>
+#include <string>
+#include <functional>
+#include <map>
+
+namespace Ioss {
+
+using FileNameGenerator = std::function<std::string(const std::string& baseFileName, unsigned step)>;
+
+enum ChangeSetFormat {
```

Comment:
Maybe `enum class`.  

I've used that some where I can.  Since this is new code it should not break existing users.

Reply:
I wanted to use `enum class` but I use bitwise composition to figure out the supported formats so I left it as `enum`